hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -181,7 +181,8 @@ private ClientMessage handleAuthenticated() {
         Connection connection = endpoint.getConnection();
         ILogger logger = clientEngine.getLogger(getClass());
 
-        endpoint.authenticated(principal, credentials, isOwnerConnection(), clientVersion);
+        clientEngine.trySetLastAuthenticationCorrelationId(principal.getUuid(), clientMessage.getCorrelationId());
```

Comment:
That was a fix for another issue. We can not backport it because we can not change operation in maintenance. Throwing exception here does not fix the issue by itself.